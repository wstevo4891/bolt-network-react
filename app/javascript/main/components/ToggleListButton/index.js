// MyListButton Container Class

import React, { Component } from 'react'
import PropTypes from 'prop-types'

import SessionList from '@services/SessionList'

import { ToggleIconButton } from '../IconButton'

class ToggleListButton extends Component {
  constructor(props) {
    super(props)
    this.state = {
      inList: null
    }
    this.list = new SessionList(props.movie, props.listName)
  }

  static propTypes = {
    movie: PropTypes.object.isRequired,
    listName: PropTypes.string.isRequired,
    updateContainer: PropTypes.func,
    buttonClass: PropTypes.string,
    text: PropTypes.string
  }

  static defaultProps = {
    updateContainer: () => true,
    buttonClass: '',
    text: ''
  }

  render() {
    const { iconOptions, buttonClass, text } = this.props
    const inList = this.state.inList

    if (inList === null) return null

    return(
      <ToggleIconButton
        status={inList}
        iconOptions={iconOptions}
        buttonClass={buttonClass}
        text={text}
        updateStatus={this.toggleList}
      />
    )
  }

  toggleList = () => {
    const inList = this.state.inList

    inList ? this.list.remove() : this.list.add()

    this.setState({ inList: !inList })

    this.props.updateContainer()
  }

  componentDidMount() {
    this.setState({
      inList: this.list.findMovie()
    })
  }
}

export default ToggleListButton
