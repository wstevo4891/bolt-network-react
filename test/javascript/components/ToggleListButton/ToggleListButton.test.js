import React from 'react'
import { shallow } from 'enzyme'

import ToggleListButton from '@components/ToggleListButton'

describe('ToggleListButton', () => {
  const mockToggle = jest.fn()

  const shallowComponent = () => shallow(
    <ToggleListButton
      movie={{ id: 1, title: 'The Avengers' }}
      iconOptions={['fa-thumbs-o-up', 'fa-thumbs-up']}
      listName="LikedList"
      updateContainer={mockToggle}
    />
  )

  let component = shallowComponent()

  it('should render correctly in "debug" mode', () => {
    expect(component).toMatchSnapshot()
  })

  it('should render with expected state', () => {
    expect(component.state().inList).toEqual(false)
  })

  describe('toggleList function', () => {
    beforeEach(() => {
      component = shallowComponent()
    })

    it('should toggle state.inList', () => {
      component.dive().dive().simulate('click')

      expect(component.state().inList).toEqual(true)

      component.dive().dive().simulate('click')

      expect(component.state().inList).toEqual(false)
    })
  })
})
