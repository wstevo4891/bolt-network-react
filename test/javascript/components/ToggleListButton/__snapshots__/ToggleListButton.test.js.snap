// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleListButton should render correctly in "debug" mode 1`] = `
<Proxy
  buttonClass=""
  iconOptions={
    Array [
      "fa-thumbs-o-up",
      "fa-thumbs-up",
    ]
  }
  status={false}
  text=""
  updateStatus={[Function]}
/>
`;
